retrievalmarket/impl/providerstates/provider_states.go

```diff
@@ -36,21 +38,33 @@ func ReceiveDeal(ctx fsm.Context, environment ProviderDealEnvironment, deal rm.P
 	}
 
 	// check that the deal parameters match our required parameters (or reject)
-	err = environment.CheckDealParams(dealProposal.PricePerByte, dealProposal.PaymentInterval, dealProposal.PaymentIntervalIncrease)
+	err = environment.CheckDealParams(dealProposal.PricePerByte,
```

Comment:
I was conflicted here between moving all decisioning to `DecideOnDeal` and leaving just the param check here. I decided an up front parameter check is okay here before the "meat" of decisioning happens, but I can be convinced otherwise.